src/core_codemods/combine_calls_base.py

```diff
@@ -0,0 +1,139 @@
+import libcst as cst
+from libcst import matchers as m
+
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from core_codemods.api import SimpleCodemod
+
+
+class CombineCallsBaseCodemod(SimpleCodemod, NameResolutionMixin):
+    combinable_funcs: list[str] = []
+    dedupilcation_attr: str = "value"
+    args_to_combine: list[int] = [0]
+    args_to_keep_as_is: list[int] = []
+
+    def leave_BooleanOperation(
+        self, original_node: cst.BooleanOperation, updated_node: cst.BooleanOperation
+    ) -> cst.CSTNode:
+        if not self.filter_by_path_includes_or_excludes(
+            self.node_position(original_node)
+        ):
+            return updated_node
+
+        for call_matcher in map(self.make_call_matcher, self.combinable_funcs):
+            if self.matches_call_or_call(updated_node, call_matcher):
+                self.report_change(original_node)
+                return self.combine_calls(updated_node.left, updated_node.right)
+
+            if self.matches_call_or_boolop(updated_node, call_matcher):
+                self.report_change(original_node)
+                return self.combine_call_or_boolop_fold_right(updated_node)
+
+            if self.matches_boolop_or_call(updated_node, call_matcher):
+                self.report_change(original_node)
+                return self.combine_boolop_or_call_fold_left(updated_node)
+
+        return updated_node
+
+    def make_call_matcher(self, func_name: str) -> m.Call:
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def check_calls_same_instance(
+        self, left_call: cst.Call, right_call: cst.Call
+    ) -> bool:
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def matches_call_or_call(
+        self, node: cst.BooleanOperation, call_matcher: m.Call
+    ) -> bool:
+        # Check for simple case: x.startswith("...") or x.startswith("...")
```

Comment:
all comments in this module related to startswith/endswith can be removed

Reply:
Sounds good, done.